autokeras/hypermodels/heads.py

```diff
@@ -173,3 +173,66 @@ def build(self, hp, inputs=None):
 
     def get_adapter(self):
         return adapters.RegressionHeadAdapter(name=self.name)
+
+
+class SegmentationHead(ClassificationHead):
+    """Classification Dense layers.
+
+    Use sigmoid and binary crossentropy for binary element segmentation.
+    Use softmax and categorical crossentropy for multi-class
+    (more than 2) segmentation. Use Accuracy as metrics by default.
+
+    The targets passing to the head would have to be tf.data.Dataset, np.ndarray,
+    pd.DataFrame or pd.Series. It can be raw labels, one-hot encoded if more than two
+    classes, or binary encoded for binary element segmentation.
+
+    The raw labels will be encoded to 0s and 1s if two classes were found, or
+    one-hot encoded if more than two classes were found.
+    One pixel only corresponds to one label.
+
+    # Arguments
+        num_classes: Int. Defaults to None. If None, it will be inferred from the
+            data.
+        loss: A Keras loss function. Defaults to use `binary_crossentropy` or
+            `categorical_crossentropy` based on the number of classes.
+        metrics: A list of Keras metrics. Defaults to use 'accuracy'.
+        dropout_rate: Float. The dropout rate for the layers.
+            If left unspecified, it will be tuned automatically.
+    """
+
+    def __init__(self,
+                 num_classes: Optional[int] = None,
+                 loss: Optional[types.LossType] = None,
+                 metrics: Optional[types.MetricsType] = None,
+                 dropout_rate: Optional[float] = None,
+                 **kwargs):
+        super().__init__(loss=loss,
+                         metrics=metrics,
+                         num_classes=num_classes,
+                         dropout_rate=dropout_rate,
+                         **kwargs)
+        self.num_classes = num_classes
+        if not self.metrics:
+            self.metrics = ['accuracy']
+        self.dropout_rate = dropout_rate
+        self.set_loss()
+
+    def set_loss(self):
+        super().set_loss()
+
+    def get_config(self):
+        return super().get_config()
+
+    def build(self, hp, inputs=None):
+        return super().build(
+            hp=hp,
+            inputs=inputs
+        )
+
+    def get_adapter(self):
+        return adapters.ClassificationHeadAdapter(name=self.name)
```

Comment:
We need a class of SegmentationHeadAdapter, which extends ClassificationHeadAdapter, in `output_adpater.py`.
It can be empty for now.
Just use `pass` as the content of the class.